model/cicd/metrics.yaml

```
@@ -0,0 +1,65 @@
+groups:
+  - id: metric.cicd.pipeline.run.duration
+    type: metric
+    metric_name: cicd.pipeline.run.duration
+    brief: 'Duration of a pipeline run grouped by pipeline and result.'
+    instrument: histogram
+    unit: "s"
+    stability: experimental
+    attributes:
+      - ref: cicd.pipeline.name
+        requirement_level: required
+      - ref: cicd.pipeline.result
+        requirement_level: required
+  - id: metric.cicd.pipeline.run.executing
+    type: metric
+    metric_name: cicd.pipeline.run.executing
+    brief: 'The number of pipeline runs currently executing.'
+    instrument: updowncounter
+    unit: "{pipeline_run}"
+    stability: experimental
+    attributes:
+      - ref: cicd.pipeline.name
+        requirement_level: required
+  - id: metric.cicd.queue.latency
+    type: metric
+    metric_name: cicd.queue.latency
+    brief: 'The duration a pipeline run takes from being triggered to the start of execution.'
+    instrument: histogram
+    unit: "s"
+    stability: experimental
+    attributes:
+      - ref: cicd.pipeline.name
+        requirement_level: recommended
+  - id: metric.cicd.queue.length
+    type: metric
+    metric_name: cicd.queue.length
+    brief: 'The number of pipeline runs waiting for their start of execution.'
+    instrument: updowncounter
+    unit: "{pipeline_run}"
+    stability: experimental
+    attributes:
+      - ref: cicd.pipeline.name
+        requirement_level: recommended
+  - id: metric.cicd.worker.count
+    type: metric
+    metric_name: cicd.worker.count
+    brief: 'The number of workers available to the CICD system and/or busy.'
```

as the workers may be down (neither available nor busy) and we can potentially add more status later, I would use a more generic definition of the metrics
```suggestion
    brief: 'The number of workers on the CI/CD system by type and status.'
```

---

I have updated the brief in dc09536748e52ae2edc25ab56b1335e171c53bbb.

---

In a2202cd19505740dc7aa10bf2a6579c1d45a143e I adapted the brief to the rename of the attribute `type` to `class`.